### ISSUE TITLE
Compiler interferance from hipify-clang

### ISSUE THREAD

**Issue by G-Ragghianti** [LLNL/spack#30711]
### Steps to reproduce

This could affect any package that depends on hipify-clang when using a compiler other than clang.
```
spack install slate+rocm amdgpu_target=gfx900 %gcc@10.2.0 ^openmpi ^mesa~llvm
```

### Error message

Errors occur when using cmake.  This is due to cmake's test programs failing when they attempt to load a header file that is included in the hipify-clang include directory.  The hipify-clang package copies the header files from clang in llvm-amdgpu into its own includes directory, and these header files are found by the gcc compiler.
```
---------------------------------------- MPI: use_mpi = True
-- Could NOT find MPI_CXX (missing: MPI_CXX_WORKS)
-- Could NOT find MPI (missing: MPI_CXX_FOUND CXX)
-- No MPI support in SLATE

```

### Information on your system

* **Spack:** 0.18.0.dev0
* **Python:** 3.6.8
* **Platform:** linux-centos7-skylake_avx512
* **Concretizer:** clingo

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by G-Ragghianti** [comment]
Causes issue https://github.com/spack/spack/issues/30596

**Comment by srekolam** [comment]
@G-Ragghianti  i will check with the team and see if they can provide any fix for the future release. right now it is a clang-based tool and we copy the include header files.

As a alternative, i have put this code in this branch.
https://github.com/spack/spack/commit/59e8257bc22e8d6de74d5d46655d3557977539f4

**Comment by srekolam** [comment]
this is fixed as part of 5.2.0 recipe update.  
@G-Ragghianti , can you please let me know if you get some chance to test it

**Comment by G-Ragghianti** [comment]
Yes this is now resolved for us.  Thanks.
